### PR TITLE
fixed declaration of inline annotation in xtend fixes #256 

### LINKED ISSUE #256: Cant declare inline annotations in xtend
```
class Foo  {
	
	def <T extends Number> m() {
  		val String s = xxxx(5)
	}
	
	@Inline("xxxx") def static String xxxx(int size) {
		null
	}
	
}
```

```
java.lang.IllegalStateException
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.appendInlineFeatureCall(FeatureCallCompiler.java:959)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.appendFeatureCall(FeatureCallCompiler.java:902)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.featureCalltoJavaExpression(FeatureCallCompiler.java:585)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler._toJavaExpression(FeatureCallCompiler.java:568)
	at org.eclipse.xtext.xbase.compiler.FeatureCallCompiler.internalToConvertedExpression(FeatureCallCompiler.java:100)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.internalToConvertedExpression(XbaseCompiler.java:342)
	at org.eclipse.xtend.core.compiler.XtendCompiler.internalToConvertedExpression(XtendCompiler.java:425)
	at org.eclipse.xtext.xbase.compiler.TypeConvertingCompiler.internalToConvertedExpression(TypeConvertingCompiler.java:104)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler._toJavaStatement(XbaseCompiler.java:587)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.doInternalToJavaStatement(XbaseCompiler.java:375)
	at org.eclipse.xtend.core.compiler.XtendCompiler.doInternalToJavaStatement(XtendCompiler.java:367)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.internalToJavaStatement(AbstractXbaseCompiler.java:473)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler._toJavaStatement(XbaseCompiler.java:394)
	at org.eclipse.xtext.xbase.compiler.XbaseCompiler.doInternalToJavaStatement(XbaseCompiler.java:349)
	at org.eclipse.xtend.core.compiler.XtendCompiler.doInternalToJavaStatement(XtendCompiler.java:367)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.internalToJavaStatement(AbstractXbaseCompiler.java:473)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.compile(AbstractXbaseCompiler.java:294)
	at org.eclipse.xtext.xbase.compiler.AbstractXbaseCompiler.compile(AbstractXbaseCompiler.java:274)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.compile(JvmModelGenerator.java:1057)
	at org.eclipse.xtend.core.compiler.XtendGenerator.compile(XtendGenerator.java:213)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateExecutableBody(JvmModelGenerator.java:1034)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._generateMember(JvmModelGenerator.java:798)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateMember(JvmModelGenerator.java:1660)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.lambda$generateMembersInBody$2(JvmModelGenerator.java:292)
	at org.eclipse.xtext.xbase.lib.ObjectExtensions.operator_doubleArrow(ObjectExtensions.java:139)
	at org.eclipse.xtext.xbase.compiler.LoopExtensions.forEach(LoopExtensions.java:35)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateMembersInBody(JvmModelGenerator.java:295)
	at org.eclipse.xtend.core.compiler.XtendGenerator.generateMembersInBody(XtendGenerator.java:573)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._generateBody(JvmModelGenerator.java:269)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateBody(JvmModelGenerator.java:1632)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.generateType(JvmModelGenerator.java:226)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator._internalDoGenerate(JvmModelGenerator.java:213)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.internalDoGenerate(JvmModelGenerator.java:1615)
	at org.eclipse.xtext.xbase.compiler.JvmModelGenerator.doGenerate(JvmModelGenerator.java:196)
	at org.eclipse.xtend.core.compiler.XtendGenerator.doGenerate(XtendGenerator.java:95)
	at org.eclipse.xtend.core.compiler.XtendGenerator.doGenerate(XtendGenerator.java:111)
	at org.eclipse.xtext.generator.GeneratorDelegate.doGenerate(GeneratorDelegate.java:43)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.handleChangedContents(ParallelBuilderParticipant.java:183)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.handleChangedContents(ParallelBuilderParticipant.java:172)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant.doGenerate(ParallelBuilderParticipant.java:343)
	at org.eclipse.xtext.builder.ParallelBuilderParticipant$1.run(ParallelBuilderParticipant.java:308)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
	at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
	at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```

### PR DESCRIPTION
**Pull request by cdietrich**
fixes #256 
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerXtendIssue256Test.xtend [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+class CompilerXtendIssue256Test	extends AbstractXtendCompilerTest {
+	@Test def void testIssue256_01() {
+		assertCompilesTo('''
+		class Foo {
+			
+			def foo() {
+		  		bar
+			}
+			
+			@Inline("\"bar\"") def static String bar() {
```
Comment (szarnekow) [inline org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerXtendIssue256Test.xtend]: Could you please add a test for imported types, too?

**Comment by cdietrich** [comment]
this wont work and would fall into the category of the general interpreter

**Comment by szarnekow** [comment]
So a fix would look similar to the one for the value, wouldn't it?

**Comment by cdietrich** [comment]
no, cause literal = xlistliteral and elements = xfeaturecall

**Comment by cdietrich** [comment]
i could imaging something like

```
 if (annotationValue instanceof JvmCustomAnnotationValue) {					
					JvmCustomAnnotationValue customAnnotationValue = (JvmCustomAnnotationValue) annotationValue;
					if (customAnnotationValue.getValues().size() == 1) {
						if (customAnnotationValue.getValues().get(0) instanceof XListLiteral) {
							EList<XExpression> elements = ((XListLiteral)customAnnotationValue.getValues().get(0)).getElements();
							for (XExpression e : elements) {
								importedTypes.add(getLightweightType(e).toTypeReference());
							}
						}
					}
				} else {
					throw new IllegalStateException("Unhandled 'imported' AnnotationValue type " + annotationValue.getClass());
				}
```

**Comment by cdietrich** [comment]
or

```
 else if ("imported".equals(annotationValue.getValueName())) {
				if (annotationValue instanceof JvmTypeAnnotationValue) {					
					JvmTypeAnnotationValue typeAnnotationValue = (JvmTypeAnnotationValue) annotationValue;
					importedTypes.addAll(typeAnnotationValue.getValues());
				} if (annotationValue instanceof JvmCustomAnnotationValue) {					
					JvmCustomAnnotationValue customAnnotationValue = (JvmCustomAnnotationValue) annotationValue;
					if (customAnnotationValue.getValues().size() == 1) {
						if (customAnnotationValue.getValues().get(0) instanceof XListLiteral) {
							EList<XExpression> elements = ((XListLiteral)customAnnotationValue.getValues().get(0)).getElements();
							for (XExpression e : elements) {
								LightweightTypeReference lightweightType = getLightweightType(e);
								if (lightweightType != null && lightweightType.isType(Class.class)) {
									importedTypes.add(lightweightType.toTypeReference());
								}
							}
						}
					}
				} else {
					throw new IllegalStateException("Unhandled 'imported' AnnotationValue type " + annotationValue.getClass());
				}
			}
```
@szarnekow do you think that could work?

**Comment by cdietrich** [comment]
updated both prs

**Comment by cdietrich** [comment]
@szarnekow good to merge now?

**Pull request merged**